src/main/java/org/jabref/gui/fieldeditors/JournalInfoOptInDialogHelper.java

```diff
@@ -0,0 +1,30 @@
+package org.jabref.gui.fieldeditors;
+
+import org.jabref.gui.DialogService;
+import org.jabref.logic.journals.JournalInformationPreferences;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.EnablementStatus;
+
+public class JournalInfoOptInDialogHelper {
+
+    /**
+     * Using the journal information data fetcher service needs to be opt-in for GDPR compliance.
+     */
+    public static boolean isJournalInfoEnabled(DialogService dialogService, JournalInformationPreferences preferences) {
+        if (preferences.getEnablementStatus() == EnablementStatus.ENABLED) {
+            return true;
+        }
+
+        if (preferences.getEnablementStatus() == EnablementStatus.DISABLED) {
+            dialogService.notify(Localization.lang("Please enable journal information fetching in Preferences > Web search"));
```

Comment:
I am unsure whether we should re-use existing localizations here.

```suggestion
            dialogService.notify(Localization.lang("Please enable journal information fetching in %0 > %1"), Localization.lang("Preferences"), Localization.lang("Web search"));
```

Reason: The translators change even for the same language. They might translate the strings differently (because they are "drive-by" translators; maybe students doing it as procrastination task during the time they are assigned to work on JabRef).

Reply:
I like the term 'drive-by translators' 😄
Updated with the suggested change